$NetBSD: patch-aa,v 1.1.1.1 2007/02/20 15:54:44 lkundrak Exp $

ffmpeg-0.4.9pre1nb2 needs this patch.
Its current version has a slightly different API, so when
ffmpeg is updated, this will have to be removed.

--- src/avformat_writer.c.orig	2007-02-20 15:02:26.000000000 +0100
+++ src/avformat_writer.c
@@ -79,7 +79,7 @@ int avfwriter_set_m4v_header(AVFWRITER* 
 		return -1;
     }
 
-    c = st->codec;
+    c = &st->codec;
     c->codec_id = CODEC_ID_MPEG4;
 	c->codec_type = CODEC_TYPE_VIDEO;	
 	
@@ -93,8 +93,8 @@ int avfwriter_set_m4v_header(AVFWRITER* 
     c->width = width;
     c->height = height;
 	
-	c->time_base.den = 30; // for 30Hz
-    c->time_base.num = 1;  // for 30Hz
+    st->time_base.den = 30; // for 30Hz
+    st->time_base.num = 1;  // for 30Hz
 	
 	if (extradatasize > 0)
 	{
@@ -125,7 +125,7 @@ int avfwriter_set_mp3_header(AVFWRITER* 
         exit(1);
     }
 
-    c = st->codec;
+    c = &st->codec;
     c->codec_id = CODEC_ID_MP3;
     c->codec_type = CODEC_TYPE_AUDIO;
 
